src/main/scala/viper/gobra/reporting/StatsCollector.scala

```diff
@@ -0,0 +1,376 @@
+  import ViperNodeType._
+
+  // Stores type info for a task
+  private[reporting] val typeInfos: Map[String, TypeInfo] = TrieMap()
+  // Maps a gobra member name to a gobra member entry
+  private[reporting] val memberMap: Map[String, GobraMemberEntry] = TrieMap()
+  // Maps a viper member name to a gobra member entry
+  private val viperMemberNameGobraMemberMap: Map[String, GobraMemberEntry] = TrieMap()
+
+  // indentation prefix for generated json
+  private val i = "  "
+  case class GobraMemberEntry(pkgDir: String,
+                              pkg: String,
+                              memberName: String,
+                              args: String,
+                              nodeType: GobraNodeType,
+                              viperMembers: Map[String, ViperMemberEntry],
+                              isTrusted: Boolean,
+                              isAbstract: Boolean) {
+
+    def asJson(p: String = ""): String = {
+      val viperMembersJson = viperMembers.values.map(_.asJson(s"$p$i$i"))
+      val dependencies = this.dependencies().map(entry => s"""$p$i$i"${entry.key}"""")
+
+      s"""$p{
+         |$p$i"id": "${this.key}",
+         |$p$i"pkgDir": "$pkgDir",
+         |$p$i"pkg": "$pkg",
+         |$p$i"name": "$pkg.$memberName",
+         |$p$i"args": "$args",
+         |$p$i"nodeType": "$nodeType",
+         |$p$i"trusted": $isTrusted,
+         |$p$i"abstract": $isAbstract,
+         |$p$i"viperMembers": [
+            |${viperMembersJson.mkString(",\n")}
+         |$p$i],
+         |$p$i"dependencies": [
+            |${dependencies.mkString(",\n")}
+         |$p$i]
+         |$p}""".stripMargin
+    }
+
+    def dependencies(): Set[GobraMemberEntry] = {
+      this.viperMembers.values
+        .flatMap(viperMember => viperMember.dependencies
+          .flatMap(dep => findGobraMemberByViperMemberName(viperMember.taskName, dep, pkgDir)))
+        .filter(_.key != this.key).toSet
+    }
+
+    def key: String = gobraMemberKey(pkgDir, pkg, memberName, args)
+  }
+
+  case class ViperMemberEntry(memberName: String,
+                              taskName: String,
+                              time: Time,
+                              nodeType: ViperNodeType,
+                              dependencies: Set[String],
+                              success: Boolean,
+                              cached: Boolean,
+                              fromImport: Boolean,
+                              hasBody: Boolean) {
+
+    def asJson(p: String = ""): String = {
+      s"""$p{
+         |$p$i"name": "${this.memberName}",
+         |$p$i"taskName": "${this.taskName}",
+         |$p$i"time": ${this.time},
+         |$p$i"nodeType": "${this.nodeType}",
+         |$p$i"success": ${this.success},
+         |$p$i"cached": ${this.cached},
+         |$p$i"fromImport": ${this.fromImport},
+         |$p$i"hasBody": ${this.hasBody}
+         |$p}""".stripMargin
+    }
+  }
+
+  case class GobraMemberInfo(pkgDir: String,
```

Comment:
why is `GobraMemberInfo` not used as a parameter of `GobraMemberEntry`? `GobraMemberEntry` seems to take a superset of parameters